frontend/website/src/components/chat/messages.svelte

```diff
@@ -0,0 +1,178 @@
+<script context="module" lang="ts">
+	export enum ChatStatus {
+		Connecting,
+		Disconnected,
+		Connected,
+	}
+</script>
+
+<script lang="ts">
+	import { pipe, subscribe, type Subscription } from "wonka";
+	import { client } from "$lib/gql";
+	import { graphql } from "$gql";
+	import { MessageType, type ChatMessage, type User } from "$gql/graphql";
+	import { onDestroy, onMount } from "svelte";
+	import type { Writable } from "svelte/store";
+
+	export let channelId: string;
+	export let onlyUserMessages: boolean = false;
+
+	onMount(() => {
+		subscribeToMessages();
+		return () => {
+			unsubscribeFromMessages();
+		};
+	});
+
+	onDestroy(() => {
+		unsubscribeFromMessages();
+	});
```

Comment:
Do we need both on destroy and the return of unmount?

Reply:
No, we don't.